stripes/public/js/combobox.bundle.js

```diff
@@ -0,0 +1,236 @@
+				items[this.current_focus].click();
+			}
+		}
+
+		if (e.key === "Escape") {
+			this.search.blur();
+
+			// Hide Drop Down
+			this.dropdown.classList.remove("show");
+			this.dropdown_menu.classList.remove("show");
+		}
+	}
+
+	set_highlight(items) {
+		this.remove_highlight(items);
+		if (this.current_focus > -1 && this.current_focus < items.length) {
+			items[this.current_focus].classList.add("active");
+			items[this.current_focus].scrollIntoView({ block: 'nearest' });
+		}
+	}
+
+	remove_highlight(items) {
+		items.forEach(item => {
+			item.classList.remove("active");
+		});
+	}
+
+	clear_message() {
+		let message = this.list_items_container.querySelector("h4");
+		if (message) {
+			message.remove();
+		}
+	}
+
+	show_message(text) {
+		const h4 = document.createElement("h4");
+		h4.innerText = text;
+		this.list_items_container.appendChild(h4);
+	}
+
+	set_value(value) {
+		this.current_value = value;
+		this.update_selected_in_view();
+	}
+
+	update_selected_in_view() {
+		for (let item of this.list_items) {
+			const item_value = item.innerText;
+			const is_selected = item.classList.contains("selected");
+
+			if (this.current_value == item_value) {
+				if(is_selected) {
+					item.classList.remove("selected");
+					item.querySelector("img")?.remove();
+					this.dropdown_button.querySelector("span").innerText = this.default_button_label;
+					this.current_value = null;
+				} else {
+					if (!item.querySelector("img")) item.insertAdjacentHTML("beforeend", `<img src="/assets/stripes/images/icons/check.svg" alt="check">`);
+					item.classList.add("selected");
+					this.dropdown_button.querySelector("span").textContent = this.current_value;
+				}
+			} else {
+				item.classList.remove("selected");
+				item.querySelector("img")?.remove();
+			}
+		}
+	}
+
+	set_dropdown_position() {
+		// Show dropdown before positioning
+		this.dropdown_menu.style.display = 'block';
+
+		const dropdown_rect = this.dropdown.getBoundingClientRect();
+		const dropdown_width = this.dropdown_menu.offsetWidth;
+		const offset = dropdown_rect.right - 26;
+
+		const space_on_right = window.innerWidth - offset;
+		const space_on_left = dropdown_rect.left;
+
+		if (space_on_right >= dropdown_width) {
+			this.dropdown_menu.style.left = '0';
+			this.dropdown_menu.style.right = 'unset';
+		} else if (space_on_left >= dropdown_width) {
+			this.dropdown_menu.style.left = 'unset';
+			this.dropdown_menu.style.right = '0';
+		} else {
+			if (space_on_right > space_on_left) {
+				this.dropdown_menu.style.left = '0';
+				this.dropdown_menu.style.right = 'unset';
+			} else {
+				this.dropdown_menu.style.left = 'unset';
+				this.dropdown_menu.style.right = '0';
+			}
+		}
+
+		// Hide the dropdown again after positioning
+		this.dropdown_menu.style.display = '';
+	}
+
+	fetch_stripes() {
+		// Add class to the container before fetching the result.
+		this.stripes_container.classList.add("loading");
+
+		setTimeout(() => {
```

Comment:
Why timeout? Why not instantly? Are you trying to debounce?

Reply:
Not exactly a debounce method, but I’ve added a slight delay because the loader also needs to be shown for a bit. Instantly showing it feels a bit abrupt.